sqlglot/dialects/dialect.py

```diff
@@ -797,3 +797,13 @@ def move_insert_cte_sql(self: Generator, expression: exp.Insert) -> str:
         expression = expression.copy()
         expression.set("with", expression.expression.args["with"].pop())
     return self.insert_sql(expression)
+
+
+def to_days_to_datediff_sql(self: Generator, expression: exp.ToDays) -> str:
+    return self.sql(
+        exp.DateDiff(
+            this=expression.this,
```

Comment:
There's a mutation going on here due to the `expression.this` having its `parent` changed.

Reply:
the approach is changing